compiler/gentype/TranslateTypeExprFromTypes.ml

```diff
@@ -253,6 +248,7 @@ let translate_constr ~config ~params_translation ~(path : Path.t) ~type_env =
   | ( (["Js"; "Dict"; "t"] | ["Dict"; "t"] | ["dict"] | ["Stdlib"; "Dict"; "t"]),
       [param_translation] ) ->
     {param_translation with type_ = Dict param_translation.type_}
+  | ["Stdlib_JSON"; "t"], [] -> {dependencies = []; type_ = unknown}
```

Comment:
Not sure why here it's `["Stdlib_JSON"; "t"]` but e.g. earlier it's `["Stdlib"; "Dict"; "t"]`.
The aliases seem to be set up in the same way. Any idea @cknitt ?

Reply:
Oh it's because of the type annotation: this one:

```res
let jsonEncodeString : JSON.t = JSON.Encode.string("hello")
```

maps to `["Stdlib"; "Dict"; "t"]`. So it might be that we need both ways, every time.

Reply:
yeah shouldn't we go the safer way just in case?
```suggestion
  | ["Stdlib_JSON"; "t"] | ["Stdlib"; "JSON"; "t"] | ["JSON"; "t"], [] -> {dependencies = []; type_ = unknown}
```

Reply:
OK I'm now allowing both, but there's still something in the hover:

<img width="394" alt="Screenshot 2025-04-04 at 12 39 46" src="https://github.com/user-attachments/assets/05cfc801-0998-4048-bad4-12e6a8e3df60" />

<img width="475" alt="Screenshot 2025-04-04 at 12 39 57" src="https://github.com/user-attachments/assets/833cb2fa-24cd-496b-bbc3-7d6f877d98c4" />

The two examples show differently in the hover.
And I guess we'd want `Stdlib` not to appear in either.
Thoughts @zth ?

Also wondering whether there's somewhere else where this shows up, the fact that we have 2 possible paths and only one at most is handled.

Reply:
that technically makes sense that the hover name is different because in the first case, `JSON.Encode.string` returns `t`, so it shows up with its internal module name while it does with its fully qualified name (`Stdlib.JSON.t`) when you use `JSON.t`.
But as you said, it should just be `JSON.t` in both cases. I guess we can special case stdlib in the extension.

Reply:
but this is a bit off-topic for this PR anyway, right?